src/v/s3/client.h

```diff
@@ -188,10 +194,15 @@ class client {
       const bucket_name& name,
       std::optional<object_key> prefix = std::nullopt,
       std::optional<object_key> start_after = std::nullopt,
-      std::optional<size_t> max_keys = std::nullopt);
-
-    ss::future<>
-    delete_object(const bucket_name& bucket, const object_key& key);
+      std::optional<size_t> max_keys = std::nullopt,
+      const ss::lowres_clock::duration& timeout
+      = http::default_connect_timeout);
+
+    ss::future<> delete_object(
+      const bucket_name& bucket,
+      const object_key& key,
+      const ss::lowres_clock::duration& timeout
+      = http::default_connect_timeout);
```

Comment:
we try hard to not use default args and have in the past gone for renaming the function or refactoring the call-site. I would advocate for refactoring all call-sites instead. It is much easier seeing this timeout *at call site* than *at definition*